src/expr/src/relation/func.rs

```diff
@@ -1119,6 +1221,34 @@ fn window_aggr<'a, I, A>(
     order_by: &[ColumnOrder],
     window_frame: &WindowFrame,
 ) -> Datum<'a>
+where
+    I: IntoIterator<Item = Datum<'a>>,
+    A: OneByOneAggr,
+{
+    let temp_storage = RowArena::new();
+    let iter = window_aggr_no_list::<I, A>(
+        input_datums,
+        &temp_storage,
+        wrapped_aggregate,
+        order_by,
+        window_frame,
+    );
+    callers_temp_storage.make_datum(|packer| {
+        packer.push_list(iter);
+    })
+}
+
+/// Like `window_aggr`, but doesn't perform the final wrapping in a list, returning an Iterator
+/// instead.
+fn window_aggr_no_list<'a: 'b, 'b, I, A>(
+    input_datums: I, // An entire window partition.
+    callers_temp_storage: &'b RowArena,
+    wrapped_aggregate: &AggregateFunc, // E.g., for `sum(...) OVER (...)`, this is the `sum(...)`.
+    // Note that this `order_by` doesn't have expressions, only `ColumnOrder`s. For an explanation,
+    // see the comment on `WindowExprType`.
```

Comment:
Please move the documentation into the function's doc comment.

Reply:
Done